ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java

```diff
@@ -127,6 +137,56 @@ public ReactImageView(
     mCallerContext = callerContext;
   }
 
+  public void setShouldNotifyLoadEvents(boolean shouldNotify) {
+    if (!shouldNotify) {
+      mControllerListener = null;
+      return;
```

Comment:
I think you actually want to set `mIsDirty = true` when listener is unset. Otherwise calling this method with `shouldNotify = false` may have no effect